examples/flutter_gallery/lib/demo/shrine/login.dart

```diff
@@ -84,8 +85,7 @@ class _LoginPageState extends State<LoginPage> {
                         borderRadius: BorderRadius.all(Radius.circular(7.0)),
                       ),
                       onPressed: () {
-                        _usernameController.clear();
-                        _passwordController.clear();
+                        Navigator.of(context, rootNavigator: true).pop();
```

Comment:
I don't really know what this is for but I suspect we don't actually need it here (or above).

Reply:
As discussed offline: I was totally wrong, we do need it!

Reply:
Since developers are likely to read this code, a comment that explains why `rootNavigator: true` is needed would be helpful.